serve.mjs

```diff
@@ -2,13 +2,36 @@ import { createServer } from "http";
 import { readFile } from "fs";
 import { spawn } from "child_process";
 import { platform } from "process";
+import readline from "readline";
+import { WebSocketServer } from "ws";
+import chalk from "chalk";
 
 // http-server exists but it is so bloated 😩
 
+let connected = false;
+const logUtils = {
+    info: (message) => {
+        logUtils.incoming(chalk.greenBright("<-- ") + message);
+    },
+    incoming: (message) => {
+        process.stdout.cursorTo(0);
```

Comment:
Should also call `process.stdout.clearLine()` before `cursorTo` in case a message has a shorter line than what was typed out.